recipe-server/client/control/state/actions/actions.js

```diff
@@ -0,0 +1,43 @@
+import {
+  ACTION_RECEIVE,
+} from '../action-types';
+
+import {
+  makeApiRequest,
+} from '../requests/actions';
+
+
+function receiveAction(action) {
+  return dispatch => {
+    dispatch({
+      type: ACTION_RECEIVE,
+      action,
+    });
+  };
+}
+
+
+export function fetchAction(name) {
+  const requestId = `fetch-${name}`;
+  return makeApiRequest(requestId, `action/${name}/`, { method: 'GET' })
+    .then(action => receiveAction(action));
+}
+
+
+function receiveAllActions(actions) {
+  return dispatch => {
+    actions.forEach(action => {
+      dispatch({
+        type: ACTION_RECEIVE,
+        action,
+      });
+    });
+  };
+}
+
+
+export function fetchAllActions() {
+  const requestId = 'fetch-all';
+  return makeApiRequest(requestId, 'action/', { method: 'GET' })
+    .then(actions => receiveAllActions(actions));
```

Comment:
I think this would be better written using async/await, like this:

```js
export async function fetchAllActions() {
  const requestId = 'fetch-actions-all';
  const actions = await makeApiRequests(requestId, 'action/', { method: 'GET' });

  for (const action of actions) {
    dispatch({
      type: ACTION_RECEIVE,
      action,
    });
  }
}
```